tests/components/blebox/test_light.py

```diff
@@ -524,3 +525,50 @@ async def test_turn_on_failure(feature, hass, config, caplog):
     assert f"Turning on '{feature_mock.full_name}' failed: Bad value 123" in str(
         info.value
     )
+
+
+async def test_wlightbox_on_effect(wlightbox, hass, config):
+    """Test light on."""
+
+    feature_mock, entity_id = wlightbox
+
+    def initial_update():
+        feature_mock.is_on = False
+
+    feature_mock.async_update = AsyncMock(side_effect=initial_update)
+    await async_setup_entity(hass, config, entity_id)
+    feature_mock.async_update = AsyncMock()
+
+    state = hass.states.get(entity_id)
+    assert state.state == STATE_OFF
+
+    def turn_on(value):
+        feature_mock.is_on = True
+        feature_mock.effect = "POLICE"
+
+    feature_mock.async_on = AsyncMock(side_effect=turn_on)
+
+    with pytest.raises(ValueError) as info:
+        await hass.services.async_call(
+            "light",
+            SERVICE_TURN_ON,
+            {"entity_id": entity_id, ATTR_EFFECT: "NOT IN LIST"},
+            blocking=True,
+        )
+
+    assert (
+        f"Turning on with effect '{feature_mock.full_name}' failed: NOT IN LIST not in effect list."
+        in str(info.value)
+    )
+
+    await hass.services.async_call(
+        "light",
+        SERVICE_TURN_ON,
+        {"entity_id": entity_id, ATTR_EFFECT: "POLICE"},
+        blocking=True,
+    )
+
+    feature_mock.async_on = AsyncMock(side_effect=turn_on)
```

Comment:
This line seems duplicate from line 549.
```suggestion
```

Reply:
Line removed.